lib/commands/up.ts

```diff
@@ -40,12 +40,23 @@ export const up = async (opts: CommandUpOptions): Promise<void> => {
       globPattern,
       globOptions
     );
-    const migrations = migrationObjs.filter(
-      (migration: MigrationObject) =>
-        appliedMigrations.find(
-          (m: MigrationModel) => m.className === migration.className
-        ) === undefined
-    );
+    const migrations = migrationObjs
+      .filter(
+        (migration: MigrationObject) =>
+          appliedMigrations.find(
+            (m: MigrationModel) => m.className === migration.className
+          ) === undefined
+      )
+      .sort((a, b): number => {
+        // sort migrations by timestamp before applying
+        const aTimestamp = Number(
+          a.className.substring(a.className.length - 13)
+        );
+        const bTimestamp = Number(
+          b.className.substring(a.className.length - 13)
+        );
+        return aTimestamp > bTimestamp ? 1 : -1;
+      });
```

Comment:
Can't we compare the complete className using `localeCompre` and not be dependent on the name format?